core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1038,9 +1039,10 @@ class ReplicaManager(val config: KafkaConfig,
     callback: ((Map[TopicPartition, Errors], Map[TopicPartition, VerificationGuard])) => Unit,
     transactionSupportedOperation: TransactionSupportedOperation
   ): Unit = {
+    val transactionLogConfig = new TransactionLogConfig(config)
```

Comment:
this is the hot path, so could you please change the condition order to avoid creating `TransactionLogConfig` too early.

Reply:
for example:
```scala
    def transactionPartitionVerificationEnable = {
      val transactionLogConfig = new TransactionLogConfig(config)
      transactionLogConfig.transactionPartitionVerificationEnable
    }
    
    // Skip verification if the request is not transactional or transaction verification is disabled.
    if (transactionalId == null || addPartitionsToTxnManager.isEmpty ||
      (!transactionSupportedOperation.supportsEpochBump && !transactionPartitionVerificationEnable)
    ) {
      callback((Map.empty[TopicPartition, Errors], Map.empty[TopicPartition, VerificationGuard]))
      return
    }
```